crates/aptos/src/move_tool/show.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::IncludedArtifactsArgs;
+use crate::common::types::{CliCommand, CliError, CliResult, CliTypedResult, MovePackageDir};
+use anyhow::Context;
+use aptos_framework::{BuildOptions, BuiltPackage};
+use aptos_types::transaction::EntryABI;
+use async_trait::async_trait;
+use clap::{Parser, Subcommand};
+
+/// Commands for showing information about compilation artifacts.
+#[derive(Subcommand)]
+pub enum ShowTool {
+    /// Compile the package and show information about the ABIs of the compiled modules.
+    Abi(Abi),
+}
+
+impl ShowTool {
+    pub async fn execute_serialized(self) -> CliResult {
+        match self {
+            Self::Abi(tool) => tool.execute_serialized().await,
+        }
+    }
+}
+
+/// Compile the package and show information about the ABIs of the compiled modules.
+///
+/// For example, this would show the function `transfer` in the module `coin`:
+///
+/// aptos move show abi --modules coin --names transfer
+///
+#[derive(Parser)]
+pub struct Abi {
+    /// If provided, only show items from the given Move modules. These should be module
+    /// names, not file paths. For example, `coin`.
+    #[clap(long, multiple_values = true)]
+    modules: Vec<String>,
+
+    /// If provided, only show items with the given names. For example, `transfer`.
+    #[clap(long, multiple_values = true)]
+    names: Vec<String>,
+
```

Comment:
How do you envision this if the ABI format changes?

Mostly because ABI here is only for entry functions, which can be confusing if something is done around view functions and others.

Reply:
Which part of this do you mean, the filtering? If the ABI changes it'll require wholesale changes in a lot of places, not sure this is special in any way?